crates/matrix-sdk-crypto/src/types/events/to_device.rs

```diff
@@ -0,0 +1,528 @@
+// Copyright 2022 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{collections::BTreeMap, fmt::Debug};
+
+use ruma::{
+    events::{
+        dummy::ToDeviceDummyEvent,
+        forwarded_room_key::ToDeviceForwardedRoomKeyEvent,
+        key::verification::{
+            accept::ToDeviceKeyVerificationAcceptEvent, cancel::ToDeviceKeyVerificationCancelEvent,
+            done::ToDeviceKeyVerificationDoneEvent, key::ToDeviceKeyVerificationKeyEvent,
+            mac::ToDeviceKeyVerificationMacEvent, ready::ToDeviceKeyVerificationReadyEvent,
+            request::ToDeviceKeyVerificationRequestEvent, start::ToDeviceKeyVerificationStartEvent,
+        },
+        room::encrypted::ToDeviceRoomEncryptedEvent,
+        room_key_request::ToDeviceRoomKeyRequestEvent,
+        secret::request::{SecretName, ToDeviceSecretRequestEvent},
+        EventContent, ToDeviceEventType,
+    },
+    serde::Raw,
+    OwnedUserId, UserId,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::{
+    value::{to_raw_value, RawValue},
+    Value,
+};
+use zeroize::Zeroize;
+
+use super::{room_key::RoomKeyEvent, secret_send::SecretSendEvent, EventType};
+use crate::types::events::from_str;
+
+/// An enum over the various to-device events we support.
+#[derive(Debug)]
+pub enum ToDeviceEvents {
```

Comment:
Since this is not a collection type (i.e. an instance only contains one event), its name should be singular, right? We could use the same naming as Ruma (`AnyToDeviceEvent`).

Reply:
I agree, but having two `AnyToDeviceEvent` types in the crate would confuse me to no end. Eventually we'll use only our customized one in the `matrix-sdk-crypto` crate and we'll rename it then.

So bear with me until we get to that point.